packages/vite/src/node/plugins/workerImportMetaUrl.ts

```diff
@@ -110,17 +110,29 @@ export function workerImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
     asSrc: true,
   }
 
+  const isIncludeWorkerImportMetaUrl = (code: string): boolean => {
+    if (
+      (code.includes('new Worker') || code.includes('new SharedWorker')) &&
+      code.includes('new URL') &&
+      code.includes(`import.meta.url`)
+    ) {
+      return true
+    }
+    return false
+  }
```

Comment:
Since this function isn't using any variables in `workerImportMetaUrlPlugin`, I think we can move this out and above the `workerImportMetaUrlPlugin` plugin as a normal `function isIncludeWorkerImportMetaUrl(code: string)`.